lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -164,20 +166,21 @@ const URLSelector = ({
 					}
 				/>
 			</Flex>
-			{!_.isEmpty(recentImages) && (
+			{recentImages.length > 0 && (
 				<Flex flexDirection="column">
 					<Txt fontSize={18}>Recent</Txt>
 					<Card
 						style={{ padding: '10px 15px' }}
-						rows={_.map(recentImages, (recent) => (
+						rows={recentImages.map((recent) => (
 							<Txt
 								key={recent}
 								onClick={() => {
 									setImageURL(recent);
 								}}
 							>
 								<span>
-									{_.last(_.split(recent, '/'))} - {recent}
+									{middleEllipsis(recent.split('/').pop() || '', 45)} -{' '}
```

Comment:
`recent.split('/').pop()` is not good enough if the url has query parameters,
`new URL(recent).pathname.split('/').pop()` would be better